localstack/services/s3/s3_listener.py

```diff
@@ -688,7 +688,7 @@ def add_accept_range_header(response):
 def is_object_expired(bucket_name: str, key: str) -> bool:
     bucket = BackendState.get_bucket(bucket_name)
     key_obj = bucket.keys.get(key)
-    if not key_obj or not key_obj._expiry:
+    if not key_obj or not getattr(key_obj, "_expiry", False):
```

Comment:
if the attribute doesn't exist anymore, then probably it doesn't make sense to try and access it in the first place, even with `getattr`. maybe the attribute moved somewhere else in moto?

Reply:
Looks like you already have similar logic here https://github.com/localstack/localstack/blob/1bbc07609287f0df1310b66e73efb42524ed9d65/localstack/services/s3/utils.py#L118-L121
I guess that it is needed only for `FakeDeleteMarker` instance.

Reply:
Right. The issue sits right there: when getting an object in a versioned bucket, a deleted key would not be deleted but be marked as a `FakeDeleteMarker`, which cannot expire and does not have the `_expiry` attribute.
You can either use the `is_key_expired` helper function here to replace the part you modified, or implement the same logic in your PR.

Reply:
Also, once fixed, you could test the change with the following test:
`tests.integration.s3.test_s3.TestS3.test_get_object_after_deleted_in_versioned_bucket`
You can remove the `xfail` as it should be fixed in both S3 providers.

Reply:
@bentsku could you please help me with understanding why `test_get_object_after_deleted_in_versioned_bucket` test [failed](https://app.circleci.com/pipelines/github/localstack/localstack/10947/workflows/73abe8ef-82a5-4461-b3b8-d2e31ed00e48/jobs/74695/tests) and how I can fix this?

Reply:
Oh sorry, I overlooked that part. The current S3 provider has some parity issues with AWS. I suggested a change, you might need to copy/paste it and run the linter again to be sure.
This change allows the snapshot verification to skip those 2 values that are returned by LocalStack but should not be there. Again, sorry for overlooking that part!
And thanks a lot again!
(If you're interested in the snapshot mechanism and how we use it in LocalStack, you can read more here: https://localstack.cloud/blog/2022-08-04-parity-explained/)

Reply:
Thank you!